### PR TITLE
Small changes to gossip related telemetry docs

### LINKED ISSUE #9820: Review telemetry docs for missing or inaccurate memberlist metrics
Our [telemetry docs page](https://www.consul.io/docs/agent/telemetry) for the memberlist `alive`, `dead` and `suspect` message rates are not super clear - we should verify but I think they are incremented every time we handle one of the message types so are the rate of receiving those types in memberlist but our telemetry page states:

> This metric counts the number of alive agents, that the agent has mapped out so far, based on the message information given by the network layer.

Just writing this so we don't forget while I'm in the middle of something else!

### PR DESCRIPTION
**Pull request by preetapan**
Addresses/Fixes #9820 

I haven't done a full sweep through memberlist/serf to make sure we aren't missing any metrics yet. First commit addresses the items brought up by @banks in #9820

### CODE REVIEW

File: website/content/docs/agent/telemetry.mdx [hunk 1]
```diff
@@ -303,17 +303,18 @@ These metrics give insight into the health of the cluster as a whole.
...
+| `consul.memberlist.msg_alive`         | This metric counts the number of alive messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | nodes / Interval                        | counter |
+| `consul.memberlist.msg_dead`          | This metric gives the number of dead messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                                | nodes / Interval                        | counter |
```
Comment (banks) [inline website/content/docs/agent/telemetry.mdx:308]:
```suggestion
| `consul.memberlist.msg_alive`         | This metric counts the number of alive messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | messages / Interval                        | counter |
| `consul.memberlist.msg_dead`          | This metric gives the number of dead messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                                | messages / Interval                        | counter |
```

File: website/content/docs/agent/telemetry.mdx [hunk 2]
```diff
@@ -303,17 +303,18 @@ These metrics give insight into the health of the cluster as a whole.
...
-| `consul.memberlist.msg_suspect`       | This metric gives the number of suspect nodes, that the agent has mapped out so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | nodes / Interval                        | counter |
+| `consul.memberlist.gossip`            | This metric measures the time taken for gossip messages to be broadcasted to a set of randomly selected nodes.                                                                                                                                                                                                                                                                                                                                 | messages / Interval                     | counter |
```
Comment (banks) [inline website/content/docs/agent/telemetry.mdx:306]:
```suggestion
| `consul.memberlist.gossip`            | This metric measures the time taken for gossip messages to be broadcasted to a set of randomly selected nodes.                                                                                                                                                                                                                                                                                                                                 | ms                  | timer |
```

File: website/content/docs/agent/telemetry.mdx [hunk 3]
```diff
@@ -303,17 +303,18 @@ These metrics give insight into the health of the cluster as a whole.
...
+| `consul.memberlist.msg_dead`          | This metric gives the number of dead messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                                | nodes / Interval                        | counter |
+| `consul.memberlist.msg_suspect`       | This metric gives the number of suspect messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | nodes / Interval                        | counter |
```
Comment (banks) [inline website/content/docs/agent/telemetry.mdx:309]:
```suggestion
| `consul.memberlist.msg_suspect`       | This metric gives the number of suspect messages, that the agent has processed so far, based on the message information given by the network layer.                                                                                                                                                                                                                                                                                              | messages / Interval                        | counter |
```

**Review by banks — APPROVED** [review]
Thanks Preetha.

I'm not sure we do need to audit actually - I realised that the one I thought was 'missing' actually was present just my search wasn't finding it initially for some reason.

I added suggestions for fixing up the units to match the descriptions too but LGTM!

**Comment by preetapan** [comment]
Thanks for the review @banks, I'll merge later today after checking Netlify preview

**Pull request merged**